api/libproject/devstart.rb

```diff
@@ -1998,6 +1997,83 @@ def create_wgs_cohort_extraction_bp_workspace(cmd_name, *args)
   :fn => ->(*args) { create_wgs_cohort_extraction_bp_workspace("create-wgs-cohort-extraction-bp-workspace", *args) }
 })
 
+def create_terra_method_snapshot(cmd_name, *args)
+  ensure_docker cmd_name, args
+
+  common = Common.new
+  op = WbOptionsParser.new(cmd_name, args)
+  op.add_option(
+    "--project [GOOGLE_PROJECT]",
+    ->(opts, v) { opts.project = v},
+    "Google project to act on (e.g. all-of-us-workbench-test). Cannot be used with --all-projects"
+  )
+  op.opts.all_projects = false
+  op.add_option(
+    "--all-projects [all-projects]",
+    ->(opts, _) { opts.all_projects = true},
+    "Create snapshot in every AoU environment. Cannot be used with --project.")
+  op.add_option(
+    "--source-git-repo [source-git-repo]",
+    ->(opts, v) { opts.source_git_repo = v},
+    "git owner/repo where the source file is located. ex. (broadinstitute/gatk)")
+  op.add_option(
+    "--source-git-path [source-git-path]",
+    ->(opts, v) { opts.source_git_path = v},
+    "git path where the source file is located, relative to the repo's root directory. ex. (scripts/variantstore/wdl/ngs_cohort_extract.wdl)")
+  op.add_option(
+    "--source-git-ref [source-git-ref]",
+    ->(opts, v) { opts.source_git_ref = v},
+    "git commit/branch/tag where the source file is located. ex. (main)")
+  op.add_option(
+    "--method-name [method-name]",
+    ->(opts, v) { opts.method_name = v},
+    "Agora method name to create snapshot in. Method Namespace will be pulled from WorkbenchConfig.wgsCohortExtraction.extractionMethodConfigurationNamespace")
+  op.add_validator ->(opts) {
+    unless (opts.source_git_repo or opts.source_git_path or opts.source_git_ref or opts.method_name)
+      common.error "all arguments must be provided"
+      raise ArgumentError
+    end
+  }
+  op.add_validator ->(opts) {
+    if (!opts.project and !opts.all_projects)
+      common.error "A project must be set or --all-projects must be true"
+      raise ArgumentError
+    end
+  }
+
+  # Use GcloudContextV2 to validate gcloud auth but we need to drop the
+  # --project argument validation that's built into the constructor
+  GcloudContextV2.validate_gcloud_auth()
+  op.parse.validate
+
+  projects = op.opts.all_projects ? ENVIRONMENTS.keys - ["local"] : [op.opts.project]
+
+  projects.each { |project|
+    flags = ([
+      ["--config-json", get_config_file(project)],
+      ["--source-git-repo", op.opts.source_git_repo],
+      ["--source-git-path", op.opts.source_git_path],
+      ["--source-git-ref", op.opts.source_git_ref],
```

Comment:
One thing that's not being captured here is the commit hash. Also there's a small concern that a commit could go in between when `staging` and `prod` are getting applied, and we wouldn't easily see a difference.

One possible solution is to resolve branch ref into a sha1 (before the project loop) before passing off the value to the CLI.